cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/PreviewHttpHandlerTest.java

```diff
@@ -15,21 +15,114 @@
+import co.cask.cdap.internal.app.preview.PreviewRuntimeService;
+import co.cask.cdap.logging.guice.LogReaderRuntimeModules;
+import co.cask.cdap.logging.guice.LoggingModules;
+import co.cask.cdap.metadata.MetadataServiceModule;
+import co.cask.cdap.metrics.guice.MetricsClientRuntimeModule;
+import co.cask.cdap.metrics.guice.MetricsHandlerModule;
+import co.cask.cdap.notifications.feeds.guice.NotificationFeedServiceRuntimeModule;
+import co.cask.cdap.notifications.guice.NotificationServiceRuntimeModule;
 import co.cask.cdap.proto.id.ApplicationId;
+import co.cask.cdap.security.authorization.AuthorizationEnforcementModule;
+import co.cask.cdap.security.guice.SecureStoreModules;
+import co.cask.cdap.store.guice.NamespaceStoreModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Assert;
+import org.junit.BeforeClass;
 import org.junit.Test;
 
 import java.util.HashSet;
 
 /**
  * Test for {@link PreviewHttpHandler}.
  */
-public class PreviewHttpHandlerTest extends AppFabricTestBase {
+public class PreviewHttpHandlerTest {
+
+  private static Injector injector;
+
+  @BeforeClass
+  public static void beforeClass() throws Throwable {
+    injector = Guice.createInjector(
+      new ConfigModule(CConfiguration.create(), new Configuration()),
+      new IOModule(),
+      new DataFabricModules().getInMemoryModules(),
+      new DataSetsModules().getStandaloneModules(),
+      new TransactionExecutorModule(),
+      new DataSetServiceModules().getInMemoryModules(),
+      new DiscoveryRuntimeModule().getInMemoryModules(),
+      new AppFabricServiceRuntimeModule().getInMemoryModules(),
+      new ServiceStoreModules().getInMemoryModules(),
+      new ProgramRunnerRuntimeModule().getInMemoryModules(),
+      new NonCustomLocationUnitTestModule().getModule(),
+      new LoggingModules().getInMemoryModules(),
+      new LogReaderRuntimeModules().getInMemoryModules(),
+      new MetricsHandlerModule(),
+      new MetricsClientRuntimeModule().getInMemoryModules(),
+      new ExploreClientModule(),
+      new NotificationFeedServiceRuntimeModule().getInMemoryModules(),
+      new NotificationServiceRuntimeModule().getInMemoryModules(),
+      new ConfigStoreModule().getInMemoryModule(),
+      new ViewAdminModules().getInMemoryModules(),
+      new StreamAdminModules().getInMemoryModules(),
+      new StreamServiceRuntimeModule().getInMemoryModules(),
+      new NamespaceStoreModule().getStandaloneModules(),
+      new MetadataServiceModule(),
+      new RemoteSystemOperationsServiceModule(),
+      new AuthorizationModule(),
+      new AuthorizationEnforcementModule().getStandaloneModules(),
+      new SecureStoreModules().getInMemoryModules(),
+      new PreviewServerModule()
+    );
+  }
+
+  private Injector getInjector() {
+    return injector;
+  }
 
   @Test
   public void testInjector() throws Exception {
```

Comment:
should we add some tests for caching? Make sure that unnecessary injectors don't get created, etc?